File: client/assets/images/generate-icons.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

// Function to create a dollar icon
function createDollarIcon(size, filename) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Background circle with gradient effect
    const centerX = size / 2;
    const centerY = size / 2;
    const radius = size * 0.4;
    
    // Create gradient
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#2E8B57');
    gradient.addColorStop(0.5, '#32CD32');
    gradient.addColorStop(1, '#90EE90');
    
    // Draw background circle
    ctx.fillStyle = gradient;
    ctx.beginPath();
    ctx.arc(centerX, centerY, radius, 0, 2 * Math.PI);
    ctx.fill();
    
    // Add border
    ctx.strokeStyle = '#228B22';
    ctx.lineWidth = size * 0.02;
    ctx.stroke();
    
    // Draw dollar sign
    ctx.fillStyle = 'white';
    ctx.font = `bold ${size * 0.6}px Arial`;
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';
    
    // Add shadow effect
    ctx.shadowColor = 'rgba(0, 0, 0, 0.3)';
    ctx.shadowBlur = size * 0.01;
    ctx.shadowOffsetX = size * 0.008;
    ctx.shadowOffsetY = size * 0.008;
    
    ctx.fillText('$', centerX, centerY);
    
    // Save the image
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(filename, buffer);
    console.log(`✅ Created ${filename} (${size}x${size})`);
}

// Create the icons
try {
    const assetsPath = path.join(__dirname);
    
    // Main app icon (512x512)
    createDollarIcon(512, path.join(assetsPath, 'icon.png'));
    
    // Adaptive icon (108x108)
    createDollarIcon(108, path.join(assetsPath, 'adaptive-icon.png'));
    
    // Favicon (32x32)
    createDollarIcon(32, path.join(assetsPath, 'favicon.png'));
    
    // Splash icon (200x200)
    createDollarIcon(200, path.join(assetsPath, 'splash-icon.png'));
    
    console.log('🎉 All dollar icons created successfully!');
    console.log('📱 Restart your Expo development server to see the new icons.');
    
} catch (error) {
    console.error('❌ Error creating icons:', error.message);
    console.log('💡 Make sure to install the canvas package:');
    console.log('   npm install canvas');
    console.log('');
    console.log('🔧 Alternative: Use the HTML file method:');
    console.log('   1. Open generate-dollar-icon.html in your browser');
    console.log('   2. Right-click on each icon and save as PNG');
    console.log('   3. Replace the existing icon files');
}

File: client/assets/images/generate_icons.py
from PIL import Image, ImageDraw, ImageFont
import os

def create_dollar_icon(size, filename):
    """Create a dollar icon with the specified size"""
    # Create image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Calculate dimensions
    center = size // 2
    radius = int(size * 0.4)
    border_width = max(1, size // 64)
    
    # Draw background circle with green color
    left = center - radius
    top = center - radius
    right = center + radius
    bottom = center + radius
    
    # Draw filled circle (background)
    draw.ellipse([left, top, right, bottom], fill=(46, 139, 87, 255))  # Sea green
    
    # Draw border
    draw.ellipse([left, top, right, bottom], outline=(34, 139, 34, 255), width=border_width)  # Forest green
    
    # Draw dollar sign
    try:
        # Try to use a system font
        font_size = int(size * 0.6)
        try:
            font = ImageFont.truetype("arial.ttf", font_size)
        except:
            try:
                font = ImageFont.truetype("C:/Windows/Fonts/arial.ttf", font_size)
            except:
                font = ImageFont.load_default()
    except:
        font = ImageFont.load_default()
    
    # Draw dollar sign
    text = "$"
    
    # Get text bounding box for centering
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    text_x = center - text_width // 2
    text_y = center - text_height // 2
    
    # Draw text with white color
    draw.text((text_x, text_y), text, fill=(255, 255, 255, 255), font=font)
    
    # Save the image
    img.save(filename, "PNG")
    print(f"✅ Created {filename} ({size}x{size})")

def main():
    """Generate all required icons"""
    try:
        # Get current directory (should be assets/images)
        current_dir = os.path.dirname(os.path.abspath(__file__))
        
        # Create icons with different sizes
        icons = [
            (512, 'icon.png'),           # Main app icon
            (108, 'adaptive-icon.png'),  # Android adaptive icon
            (32, 'favicon.png'),         # Web favicon
            (200, 'splash-icon.png'),    # Splash screen icon
        ]
        
        for size, filename in icons:
            filepath = os.path.join(current_dir, filename)
            create_dollar_icon(size, filepath)
        
        print("🎉 All dollar icons created successfully!")
        print("📱 Restart your Expo development server to see the new icons.")
        
    except ImportError:
        print("❌ PIL (Pillow) is not installed.")
        print("💡 Install it with: pip install Pillow")
        print("")
        print("🔧 Alternative methods:")
        print("   1. Open generate-dollar-icon.html in your browser")
        print("   2. Right-click on each icon and 'Save image as' PNG")
        print("   3. Replace the existing icon files")
    except Exception as e:
        print(f"❌ Error creating icons: {e}")

if __name__ == "__main__":
    main()
